### ISSUE TITLE
switch-client from run-shell cancels repeat

### ISSUE THREAD

**Issue by philsauce** [tmux/tmux#3733]
### Issue description

The `bind-key`'s repeat functionality (`-r`) doesn't work when using `tmux switch-client` from inside a `run-shell` command.

EG. With the prefix table active, this key binding will switch the session each time the command key `<` is entered in quick succession:
```
bind-key -r -T prefix < switch-client -p
```

While this binding will only switch the session once for the same key sequence:
```
bind-key -r -T prefix < run-shell 'tmux switch-client -p'
```

### Required information

Please provide the following information:

* tmux version (`tmux -V`). tmux next-3.4
* Platform (`uname -sp`). Linux unknown
* $TERM inside and outside of tmux (`echo $TERM`). tmux-256color, xterm-256color

**Comment by nicm** [comment]
Yes, the information that the source key binding was repeated is lost. I don't see a practical way to preserve it. I guess we could add a flag to `switch-client` so you could explicitly tell it not to reset the key table...

Is this an actual problem for some use case?

**Comment by philsauce** [comment]
Example use case:

I have 2 functions to handle window alerts in tmux. One tracks window IDs with alerts. The other (the intended target for `bind-key -r`) switches to the window with the most recent alert (like `next-window -a`, but based on the latest alert rather than sequential order). Because it tests multiple conditions and has side effects (like clearing the most recent window ID from the list, or displaying an error if no applicable window is found) it’s stored in a shell script, and invoked via `run-shell`.

**Comment by nicm** [comment]
Try this [sc_r.diff.txt](https://github.com/user-attachments/files/23494610/sc_r.diff.txt) and then `switch-client -R` in your script and should not switch back to the `root` table.
